controllers/eventingauth_controller.go

```diff
@@ -166,7 +160,10 @@ func (r *eventingAuthReconciler) Reconcile(ctx context.Context, req ctrl.Request
 				RequeueAfter: r.errorRequeuePeriod,
 			}, createAppErr
 		}
-		logger.Info("Created IAS application secret")
+		logger.Info("Successfully created application secret on SKR")
+
+		// Because the application secret is created on the SKR, we can delete it from the cache.
+		delete(r.existingIasApplications, cr.Name)
```

Comment:
This case can always happen, since we can never rely on the IAS app being in the cache, since it is lost with every restart. So I think it's not worse than before.